test/extended/util/disruption/disruption.go

```diff
@@ -122,45 +120,45 @@ func runChaosmonkey(
 			TestData:        testData,
 			framework:       f,
 			test:            t,
-			testReport:      testCase,
+			testName:        displayName,
+			className:       "disruption_tests",
 			testSuiteReport: testSuite,
 		}
 		cm.Register(cma.Test)
 	}
 
 	start := time.Now()
 	defer func() {
-		testSuite.Update()
-		testSuite.Time = time.Since(start).Seconds()
-
-		// if the test fails and all failures are described as "Flake", create a second
-		// test case that is listed as success so the test is properly marked as flaky
-		for _, testCase := range testSuite.TestCases {
-			allFlakes := len(testCase.Failures) > 0 && len(testCase.Errors) == 0 && len(testCase.Skipped) == 0
-			for _, failure := range testCase.Failures {
-				if failure.Type == "Flake" {
-					failure.Type = "Failure"
-				} else {
-					allFlakes = false
-				}
+
+		// Calculate NumFailed and NumSkipped
+		for _, tc := range testSuite.TestCases {
+			testSuite.NumTests++
+			if tc.FailureOutput != nil {
+				testSuite.NumFailed++
 			}
-			if allFlakes {
-				testSuite.TestCases = append(testSuite.TestCases, &junit.TestCase{
-					Name:      testCase.Name,
-					Classname: testCase.Classname,
-					Time:      testCase.Time,
-				})
+			if tc.SkipMessage != nil {
+				testSuite.NumSkipped++
 			}
 		}
 
+		testSuite.Duration = time.Since(start).Seconds()
+
 		if framework.TestContext.ReportDir != "" {
-			fname := filepath.Join(framework.TestContext.ReportDir, fmt.Sprintf("junit_%s_%d.xml", testSuite.Package, time.Now().Unix()))
+			timeSuffix := fmt.Sprintf("_%s", time.Now().UTC().Format("20060102-150405"))
+
+			fname := filepath.Join(framework.TestContext.ReportDir, fmt.Sprintf("junit_%s_%s.xml", packageName, timeSuffix))
 			f, err := os.Create(fname)
 			if err != nil {
+				// TODO: no logging?
```

Comment:
Would a log entry here help in the case we had a failure and wanted to look into it?  Don't know how useful it would be so not a hard ask.